schematic/schemas/data_model_validator.py

```diff
@@ -27,6 +27,30 @@
 """
 
 
+@dataclass
+class Node:
+    """A node in graph from the data model."""
+
+    name: Any
+    """Name of the node."""
+
+    fields: dict
+    """Fields of the node"""
+
+    def get_node_display_name(self) -> str:
+        """Gets the display name from the nodes fields if it exists
+
+        Raises:
+            ValueError: If the fields don't contain 'displayName'
+
+        Returns:
+            (str): The display name of the node
+        """
+        if "displayName" not in self.fields:
+            raise ValueError(f"Node: {str(self.name)} missing displayName field")
+        return str(self.fields["displayName"])
```

Comment:
An alternative that could possibly be done here is to create an attribute on `Node` for `display_name` that gets set in `__post__init__`:
https://docs.python.org/3/library/dataclasses.html#post-init-processing

The idea is that at the time of creation of the class instance we can apply the validation then and there. Then all downstream logic knows the data has been validated to exist in the way we're expecting it to.

What you have here works, the different being when an exception can be raised, and where it's being raised. IMO If every node is expected to have a `displayName` in `fields`, it might make sense to raise the exception as soon as possible (Ie: When the `Node` instance is being constructed`).

That way the code blows up earlier; you can have error handling further up. The way it is now means that anything calling `get_node_display_name` may need to add it's error handling of how to deal with a ValueError exception.

Reply:
If we did that, we'd proably want to follow that concept to its logical conclusion, and do the same for The DatModelValidator class as well.

Currently:

```
class DataModelValidator:  # pylint: disable=too-few-public-methods
    """
    Check for consistency within data model.
    """

    def __init__(
        self,
        graph: nx.MultiDiGraph,
    ):
        """
        Args:
            graph (nx.MultiDiGraph): Graph representation of the data model.
        """
        self.graph = graph
        self.dmr = DataModelRelationships()


    def _get_node_info(self) -> list[Node]:
        """Gets the complete node information form the graph

        Returns:
            list[Node]: A list of nodes from the graph
        """
        return [Node(node[0], node[1]) for node in self.graph.nodes(data=True)]
```

proposed:

```
class DataModelValidator:  # pylint: disable=too-few-public-methods
    """
    Check for consistency within data model.
    """

    def __init__(
        self,
        graph: nx.MultiDiGraph,
    ):
        """
        Args:
            graph (nx.MultiDiGraph): Graph representation of the data model.
        """
        self.graph = graph
        self.node_info = [Node(node[0], node[1]) for node in self.graph.nodes(data=True)]
        self.dmr = DataModelRelationships()

```

Reply:
I like the idea!